worlds/ufo50/games/vainger/rules.py

```diff
@@ -0,0 +1,248 @@
+    player = world.player
+    if not state.has_from_list_unique([heat_mod, multi_mod, pulse_mod, force_mod], player, difficulty):
+        return False
+    if (difficulty == 4) and not state.has(stabilizer, player):
+        return False
+    shield_upgrades_required = [0, 0, 5, 10, 15][difficulty]
+    return (state.count(shield_upgrade, player) >= shield_upgrades_required)
+
+# can the player tank two hits from spikes without spike-ng? (either there and back through one layer, or two hits in and zero out)
+# TODO: option for this
+def spike_tank(state: CollectionState, world: "UFO50World") -> bool:
+    player = world.player
+    # without magmatek, there's no way to have enough shield to cross the spikes in both directions, so this isn't logical.
+    if not state.has(heat_mod, player):
+        return False
+    shield_upgrades_required = 6 # 105 shield, enough to take two hits with magmatek
+    return (state.count(shield_upgrade, player) >= shield_upgrades_required)
+
+def create_rules(world: "UFO50World", regions: Dict[str, Region]) -> None:
+    player = world.player
+
+    # name regions for convenience
+    latomc6 = regions["Vainger - LatomC6 Genepod"]
+    latomc9 = regions["Vainger - LatomC9 Genepod"]
+    latomd3 = regions["Vainger - LatomD3 Genepod"]
+    latomd5 = regions["Vainger - LatomD5 Genepod"]
+    latomf5 = regions["Vainger - LatomF5 Genepod"]
+    latomf7 = regions["Vainger - LatomF7 Genepod"]
+    latomi4 = regions["Vainger - LatomI4 Genepod"]
+    latomd6area = regions["Vainger - LatomD6 Area"]
+
+    thetaa4 = regions["Vainger - ThetaA4 Genepod"] 
+    thetae9 = regions["Vainger - ThetaE9 Genepod"] 
+    thetaf5 = regions["Vainger - ThetaF5 Genepod"] 
+    thetaf6 = regions["Vainger - ThetaF6 Genepod"] 
+    thetai7 = regions["Vainger - ThetaI7 Genepod"] 
+    thetai9 = regions["Vainger - ThetaI9 Genepod"] 
+    thetac8loc = regions["Vainger - ThetaC8 Location"]
+    thetac10loc = regions["Vainger - ThetaC10 Location"]
+    
+    verdea1 = regions["Vainger - VerdeA1 Genepod"]
+    verdee1 = regions["Vainger - VerdeE1 Genepod"]
+    verdee6 = regions["Vainger - VerdeE6 Genepod"]
+    verdei7 = regions["Vainger - VerdeI7 Genepod"]
+    verdei9 = regions["Vainger - VerdeI9 Genepod"]
+    verdeswarea = regions["Vainger - VerdeSW Area"]
+    verdeh7loc = regions["Vainger - VerdeH7 Location"]
+    
+    control = regions["Vainger - Control Genepod"] 
+
+    # ThetaF5 is the starting genepod
+    regions["Vainger - Menu"].connect(thetaf5)
+
+    thetaf5.connect(thetai7)
+    # whether this hell run should be in logic itemless is a big question. it's possible, and it'll expand the possibilities
+    # for heat mod placement a lot, but it's tricky and doing it every single time in sphere 1 could get old fast.
+    # currently this is considered logical itemless
+    thetaf5.connect(thetaa4,
+                    rule = lambda state: state.has(heat_mod, player) or hell_run(0, False, state, world)) #itemless hell run
```

Comment:
```suggestion
    thetaf5.connect(thetaa4,
                    rule=lambda state: state.has(heat_mod, player) or hell_run(0, False, state, world)) #itemless hell run
```
In python, you're supposed to not have spaces before and after the = if you're explicitly naming the parameter with it